nixos/modules/services/misc/sonarr.nix

```diff
@@ -48,26 +82,41 @@ in
       "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -"
     ];
 
-    systemd.services.sonarr = {
-      description = "Sonarr";
-      after = [ "network.target" ];
-      wantedBy = [ "multi-user.target" ];
+    systemd.services.sonarr =
+      let
+        # add empty ApiKey so it can be replaced afterwards
+        configContent =
+          settingsCombined // (lib.optionalAttrs (cfg.apiKeyFile != "") { ApiKey = "#APIKEY#"; });
```

Comment:
Are the hashes around APIKEY something upstream uses? If not I think `@` is a more common "wrapper" for strings to be replaced.

Reply:
done